arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/Schedule.kt

```diff
@@ -192,21 +212,48 @@ sealed class Schedule<Input, Output> {
   /**
    * Conditionally checks on both the input and the output whether or not to continue.
    */
-  abstract fun <A : Input> check(pred: suspend (A, Output) -> Boolean): Schedule<A, Output>
+  abstract fun <A : Input> check(pred: suspend (input: A, output: Output) -> Boolean): Schedule<A, Output>
 
   /**
    * Inverts the decision of a schedule.
    */
   abstract operator fun not(): Schedule<Input, Output>
 
-  /**
-   * Combines with another schedule by combining the result and the delay of the [Decision] with the functions [f] and [g]
-   */
-  abstract fun <A : Input, B> combineWith(
+  @Deprecated(
+    "combineWith is deprecated in favor of combineNanos and combine using Double/kotlin.time.Duration instead of arrow.fx.coroutines.Duration.",
+    ReplaceWith(
+      "combine(other, f, { a, b -> g(a.nanoseconds, b.nanoseconds).nanoseconds }, ::Pair)",
+      "arrow.fx.coroutines.nanoseconds"
+    )
+  )
+  fun <A : Input, B> combineWith(
     other: Schedule<A, B>,
     f: (Boolean, Boolean) -> Boolean,
     g: (Duration, Duration) -> Duration
-  ): Schedule<A, Pair<Output, B>>
+  ): Schedule<A, Pair<Output, B>> =
+    combineNanos(other, f, { a, b -> g(a.toLong().oldNanoseconds, b.toLong().oldNanoseconds).nanoseconds.toDouble() }, ::Pair)
+
+  /**
+   * Combines with another schedule by combining the result and the delay of the [Decision] with the functions [f], [g] and a [zip] function
+   */
+  @ExperimentalTime
+  fun <A : Input, B, C> combine(
+    other: Schedule<A, B>,
+    zipContinue: (cont: Boolean, otherCont: Boolean) -> Boolean,
+    zipDuration: (duration: kotlin.time.Duration, otherDuration: kotlin.time.Duration) -> kotlin.time.Duration,
```

Comment:
nit: you could import this with an alias like `kotlinDuration` or something like that